# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/base/nsDocLoader.h
# Commit: 8303383dd95b
# Full Hash: 8303383dd95b60e0b23d961bc397f670b2fd0383
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-08-15 09:46:49
# Regressor Bug: 1562292
# File Overlap Count: 1
# Description:
#   Bug 1562292: Part 2d - Operate on BrowsingContext in most of OpenWindowInternal. r=nika
#   
#   This allows us to actually correctly load URIs into and return BCs for
#   cross-process named targets. It still leaves a lot of code which needs to
#   operate directly on DocShells, but it only applies to new windows, which
# ==============================================================================

diff -r b47a29dc541a -r 8303383dd95b uriloader/base/nsDocLoader.h
--- a/uriloader/base/nsDocLoader.h	Fri Jun 28 14:34:58 2019 -0700
+++ b/uriloader/base/nsDocLoader.h	Thu Aug 01 17:08:54 2019 -0700
@@ -366,4 +366,8 @@
 
 NS_DEFINE_STATIC_IID_ACCESSOR(nsDocLoader, NS_THIS_DOCLOADER_IMPL_CID)
 
+static inline nsISupports* ToSupports(nsDocLoader* aDocLoader) {
+  return static_cast<nsIDocumentLoader*>(aDocLoader);
+}
+
 #endif /* nsDocLoader_h__ */
